browser/src/Services/FileSystemWatcher/index.ts

```diff
@@ -42,11 +43,13 @@ export class FileSystemWatcher {
         // to avoid a flurry of events when the watcher is initialised
         this._watcher.on("ready", () => {
             this._attachEventListeners()
-        })
 
-        this._workspace.onDirectoryChanged.subscribe(newDirectory => {
-            this.unwatch(this._activeWorkspace)
-            this.watch(newDirectory)
+            if (fileOrFolder === this._activeWorkspace) {
```

Comment:
This actually might be a _bit_ fragile, if we decide to create a `FileSystemWatcher` that is not coupled to the workspace, but happened to be watching the same folder - it might not be obvious that it's going to switch directories underneath if the workspace changes.

We could consider 'inverting' this dependency - meaning, instead of the `FileSystemWatcher` being dependent on the Workspace, we switch gears - and make the Workspace (or explorer) own an instance of `FileSystemWatcher`. Then, in either Workspace / Explorer, we listen to the event there - and call `watch`/`unwatch` from there - that way the `FileSystemWatcher` isn't ['coupled' ](https://en.wikipedia.org/wiki/Coupling_(computer_programming)). 

That makes the `FileSystemWatcher` simpler, since now it doesn't have to care about the workspace at all!